test/utils/bytecode.hpp

```diff
@@ -163,4 +163,35 @@ inline bytecode sstore(bytecode index, bytecode value)
 inline bytecode sload(bytecode index)
 {
     return index + OP_SLOAD;
+}
+
+
+inline std::string decode(bytes_view bytecode, evmc_revision rev)
+{
+    auto s = std::string{"bytecode{}"};
+    const auto names = evmc_get_instruction_names_table(rev);
+    for (auto it = bytecode.begin(); it != bytecode.end(); ++it)
+    {
+        const auto opcode = *it;
+        if (const auto name = names[opcode]; name)
+        {
+            s += std::string{" + OP_"} + name;
+
+            if (opcode >= OP_PUSH1 && opcode <= OP_PUSH32)
+            {
+                auto push_data = std::string{};
+                auto push_data_size = opcode - OP_PUSH1 + 1;
+                while (push_data_size-- && ++it != bytecode.end())
+                    push_data += to_hex({&*it, 1});
```

Comment:
Would be nice to have `to_hex` overload for a single byte parameter

Reply:
On the other hand, couldn't you do something like
```
auto push_data_start = ++it;
push_data_size = std::min(push_data_size, bytecode.end() - push_data_start);
push_data += to_hex({&*push_data_start, push_data_size});
it += push_data_size;
```
maybe too complicated...

Reply:
Great change actually. Allowed me to remove the additional iterator end check.